compiler/modules/types/core.bal

```diff
@@ -170,12 +170,13 @@ function cellAtomType(Atom atom) returns CellAtomicType {
     return <CellAtomicType>(<TypeAtom>atom).atomicType;
 }
 
-// See memoSubtypeIsEmpty for what these mean.
-type MemoEmpty boolean|"provisional"|();
+// See memoSubtypeIsFiniteAndEmpty for what these mean.
+type MemoEmpty boolean|"loop"|"provisional"|();
 
 type BddMemo record {|
     readonly Bdd bdd;
     MemoEmpty empty = ();
+    boolean finite = true;
```

Comment:
Why is this a separate field rather than another value for `empty`?

Reply:
fixed.